docs/sources/flow/reference/components/discovery.file.md

```diff
@@ -0,0 +1,163 @@
+---
+title: discovery.file
+---
+
+# discovery.file
+
+> **NOTE:** In `v0.35.0` of the Grafana Agent, the `discovery.file` component was renamed to [local.file_match][]. From `v0.35.0` onwards, the `discovery.file` 
+> component behaves as documented here.
```

Comment:
Maybe specify here briefly what local.file_match does, and why they may want to go there instead.